### ISSUE TITLE
var-naming[no-role-prefix] is not detected in a role's `vars` files

### ISSUE THREAD

**Issue by cavcrosby** [ansible/ansible-lint#3714]
##### Summary

var-naming[no-role-prefix] is not detected in a role's `vars` files (YAML that exists under the `defaults` and `vars` directories).

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```shell
ansible-lint --version
ansible-lint 6.18.0 using ansible-core:2.15.3 ansible-compat:4.1.9 ruamel-yaml:None ruamel-yaml-clib:None
```

- ansible installation method: `pip`
- ansible-lint installation method: `pip`

###### Setup
```shell (paste below)
mkdir -p ./roles/test/{defaults,vars}

cat << _EOF_ > "./roles/test/defaults/main.yml"
---
foo: bar
_EOF_


cat << _EOF_ > "./roles/test/vars/main.yml"
---
bar: baz
_EOF_
```

###### Run
```shell
ansible-lint --offline
```

##### Desired Behavior

I would expect that the var-naming[no-role-prefix] rule to be detected/violated in the role's `vars` files tested.

```shell
WARNING  Listing 2 violation(s) that are fatal
var-naming[no-role-prefix]: Variables names from within roles should use test_ as a prefix. (vars: foo)
roles/test/defaults/main.yml:2

var-naming[no-role-prefix]: Variables names from within roles should use test_ as a prefix. (vars: bar)
roles/test/vars/main.yml:2

Read documentation for instructions on how to ignore specific rule violations.

                    Rule Violation Summary                     
 count tag                        profile rule associated tags 
     2 var-naming[no-role-prefix] basic   idiom                

Failed: 2 failure(s), 0 warning(s) on 5 files. Last profile that met the validation criteria was 'min'.
```

##### Actual Behavior

The actual behavior is that the var-naming[no-role-prefix] rule is not detected/violated in the role's `vars` files tested.

###### Without verbosity
```shell

Passed: 0 failure(s), 0 warning(s) on 5 files. Last profile that met the validation criteria was 'production'.
```

###### With verbosity
```shell
DEBUG    Logging initialized to level 10
INFO     Identified /home/conner/git/playground as project root due .git directory.
DEBUG    Options: Options(cache_dir=PosixPath('/home/conner/.cache/ansible-compat/f3cc87'), colored=False, configured=True, cwd=PosixPath('/home/conner/git/playground/vagrant/ansible'), display_relative_path=True, exclude_paths=['.cache', '.git', '.hg', '.svn', '.tox'], format=None, lintables=[], list_rules=False, list_tags=False, write_list=None, parseable=False, quiet=0, rulesdirs=[PosixPath('/home/conner/.pyenv/versions/playground/lib/python3.9/site-packages/ansiblelint/rules')], skip_list=[], tags=[], verbosity=4, warn_list=['experimental', 'jinja', 'fqcn'], mock_filters=[], mock_modules=[], mock_roles=[], loop_var_prefix=None, only_builtins_allow_collections=[], only_builtins_allow_modules=[], var_naming_pattern=None, offline=True, project_dir='/home/conner/git/playground', extra_vars=None, enable_list=[], skip_action_validation=True, strict=False, rules={}, profile=None, task_name_prefix='{stem} | ', sarif_file=None, config_file=None, generate_ignore=False, rulesdir=[], use_default_rules=False, version=False, list_profiles=False, ignore_file=None, max_tasks=100, max_block_depth=20)
DEBUG    CWD: /home/conner/git/playground/vagrant/ansible
INFO     Set ANSIBLE_LIBRARY=/home/conner/.cache/ansible-compat/0cb87f/modules:/home/conner/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/conner/.cache/ansible-compat/0cb87f/collections:/home/conner/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/conner/.cache/ansible-compat/0cb87f/roles:roles:/home/conner/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
DEBUG    Effective yamllint rules used: {'anchors': {'level': 'error', 'forbid-undeclared-aliases': True, 'forbid-duplicated-anchors': False, 'forbid-unused-anchors': False}, 'braces': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 1, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'brackets': {'level': 'error', 'forbid': False, 'min-spaces-inside': 0, 'max-spaces-inside': 0, 'min-spaces-inside-empty': -1, 'max-spaces-inside-empty': -1}, 'colons': {'level': 'error', 'max-spaces-before': 0, 'max-spaces-after': 1}, 'commas': {'level': 'error', 'max-spaces-before': 0, 'min-spaces-after': 1, 'max-spaces-after': 1}, 'comments': {'level': 'warning', 'require-starting-space': True, 'ignore-shebangs': True, 'min-spaces-from-content': 1}, 'comments-indentation': False, 'document-end': False, 'document-start': False, 'empty-lines': {'level': 'error', 'max': 2, 'max-start': 0, 'max-end': 0}, 'empty-values': False, 'float-values': False, 'hyphens': {'level': 'error', 'max-spaces-after': 1}, 'indentation': {'level': 'error', 'spaces': 'consistent', 'indent-sequences': True, 'check-multi-line-strings': False}, 'key-duplicates': {'level': 'error'}, 'key-ordering': False, 'line-length': {'level': 'error', 'max': 160, 'allow-non-breakable-words': True, 'allow-non-breakable-inline-mappings': False}, 'new-line-at-end-of-file': {'level': 'error'}, 'new-lines': {'level': 'error', 'type': 'unix'}, 'octal-values': {'forbid-implicit-octal': True, 'forbid-explicit-octal': True, 'level': 'error'}, 'quoted-strings': False, 'trailing-spaces': {'level': 'error'}, 'truthy': {'level': 'warning', 'allowed-values': ['true', 'false'], 'check-keys': True}}
INFO     Loading ignores from .gitignore
INFO     Excluded: .vagrant
INFO     Excluded: Makefile
INFO     Excluded: ansible.log
INFO     Excluded: foo.cfg
INFO     Excluded: vagrant.log
DEBUG    Added role: roles/test (role)
INFO     Loading ignores from .gitignore
INFO     Excluded: .vagrant
INFO     Excluded: Makefile
INFO     Excluded: ansible.log
INFO     Excluded: foo.cfg
INFO     Excluded: vagrant.log
DEBUG    data set to None for .gitignore due to being '' (unknown) kind.
DEBUG    data set to None for Vagrantfile due to being '' (unknown) kind.
INFO     Set ANSIBLE_LIBRARY=/home/conner/.cache/ansible-compat/0cb87f/modules:/home/conner/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/conner/.cache/ansible-compat/0cb87f/collections:/home/conner/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/conner/.cache/ansible-compat/0cb87f/roles:roles:/home/conner/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Executing syntax check on role roles/test (0.35s)
INFO     Executing syntax check on playbook playbook.yml (0.39s)
DEBUG    Examining roles/test of type role
DEBUG    Examining roles/test/defaults/main.yml of type vars
DEBUG    Examining playbook.yml of type playbook
DEBUG    Examining roles/test/vars/main.yml of type vars

DEBUG    Determined rule-profile order: {'internal-error': (0, 'min'), 'load-failure': (1, 'min'), 'parser-error': (2, 'min'), 'syntax-check': (3, 'min'), 'command-instead-of-module': (4, 'basic'), 'command-instead-of-shell': (5, 'basic'), 'deprecated-bare-vars': (6, 'basic'), 'deprecated-local-action': (7, 'basic'), 'deprecated-module': (8, 'basic'), 'inline-env-var': (9, 'basic'), 'key-order': (10, 'basic'), 'literal-compare': (11, 'basic'), 'jinja': (12, 'basic'), 'no-free-form': (13, 'basic'), 'no-jinja-when': (14, 'basic'), 'no-tabs': (15, 'basic'), 'partial-become': (16, 'basic'), 'playbook-extension': (17, 'basic'), 'role-name': (18, 'basic'), 'schema': (19, 'basic'), 'name': (20, 'basic'), 'var-naming': (21, 'basic'), 'yaml': (22, 'basic'), 'name': (23, 'moderate'), 'name': (24, 'moderate'), 'name': (25, 'moderate'), 'spell-var-name': (26, 'moderate'), 'avoid-implicit': (27, 'safety'), 'latest': (28, 'safety'), 'package-latest': (29, 'safety'), 'risky-file-permissions': (30, 'safety'), 'risky-octal': (31, 'safety'), 'risky-shell-pipe': (32, 'safety'), 'galaxy': (33, 'shared'), 'ignore-errors': (34, 'shared'), 'layout': (35, 'shared'), 'meta-incorrect': (36, 'shared'), 'meta-no-tags': (37, 'shared'), 'meta-video-links': (38, 'shared'), 'meta-version': (39, 'shared'), 'meta-runtime': (40, 'shared'), 'no-changed-when': (41, 'shared'), 'no-changelog': (42, 'shared'), 'no-handler': (43, 'shared'), 'no-relative-paths': (44, 'shared'), 'max-block-depth': (45, 'shared'), 'max-tasks': (46, 'shared'), 'unsafe-loop': (47, 'shared'), 'avoid-dot-notation': (48, 'production'), 'sanity': (49, 'production'), 'fqcn': (50, 'production'), 'import-task-no-when': (51, 'production'), 'meta-no-dependencies': (52, 'production'), 'single-entry-point': (53, 'production'), 'use-loop': (54, 'production')}
Passed: 0 failure(s), 0 warning(s) on 6 files. Last profile that met the validation criteria was 'production'.
```
